npm create astro@latest looks for... pnpm?

### What version of `astro` are you using?

latest

### Are you using an SSR adapter? If so, which one?

No

### What package manager are you using?

npm

### What operating system are you using?

Windows

### What browser are you using?

Chrome

### Describe the Bug

On running `npm create astro@latest`, 

```
file:///C:/Users/user/AppData/Local/npm-cache/_npx/abdb4b598af046c4/node_modules/execa/lib/error.js:59
                error = new Error(message);
                        ^

Error: Command failed with exit code 1: pnpm config get registry
'pnpm' is not recognized as an internal or external command,
operable program or batch file.
    at makeError (file:///C:/Users/user/AppData/Local/npm-cache/_npx/abdb4b598af046c4/node_modules/execa/lib/error.js:59:11)
    at handlePromise (file:///C:/Users/user/AppData/Local/npm-cache/_npx/abdb4b598af046c4/node_modules/execa/index.js:119:26)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async getRegistry (file:///C:/Users/user/AppData/Local/npm-cache/_npx/abdb4b598af046c4/node_modules/create-astro/dist/index.js:228:31)
    at async file:///C:/Users/user/AppData/Local/npm-cache/_npx/abdb4b598af046c4/node_modules/create-astro/dist/index.js:279:20 {
  shortMessage: 'Command failed with exit code 1: pnpm config get registry',
  command: 'pnpm config get registry',
  escapedCommand: 'pnpm config get registry',
  exitCode: 1,
  signal: undefined,
  signalDescription: undefined,
  stdout: '',
  stderr: "'pnpm' is not recognized as an internal or external command,\r\n" +
    'operable program or batch file.',
  failed: true,
  timedOut: false,
  isCanceled: false,
  killed: false
}

Node.js v18.16.0
npm ERR! code 1
npm ERR! path D:\htmlcss
npm ERR! command failed
npm ERR! command C:\Windows\system32\cmd.exe /d /s /c create-astro

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\user\AppData\Local\npm-cache\_logs\2023-06-12T16_42_49_111Z-debug-0.log
 htmlcss  npm create astro@latest
file:///C:/Users/user/AppData/Local/npm-cache/_npx/abdb4b598af046c4/node_modules/execa/lib/error.js:59
                error = new Error(message);
                        ^

Error: Command failed with exit code 1: pnpm config get registry
'pnpm' is not recognized as an internal or external command,
operable program or batch file.
    at makeError (file:///C:/Users/user/AppData/Local/npm-cache/_npx/abdb4b598af046c4/node_modules/execa/lib/error.js:59:11)
    at handlePromise (file:///C:/Users/user/AppData/Local/npm-cache/_npx/abdb4b598af046c4/node_modules/execa/index.js:119:26)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async getRegistry (file:///C:/Users/user/AppData/Local/npm-cache/_npx/abdb4b598af046c4/node_modules/create-astro/dist/index.js:228:31)
    at async file:///C:/Users/user/AppData/Local/npm-cache/_npx/abdb4b598af046c4/node_modules/create-astro/dist/index.js:279:20 {
  shortMessage: 'Command failed with exit code 1: pnpm config get registry',
  command: 'pnpm config get registry',
  escapedCommand: 'pnpm config get registry',
  exitCode: 1,
  signal: undefined,
  signalDescription: undefined,
  stdout: '',
  stderr: "'pnpm' is not recognized as an internal or external command,\r\n" +
    'operable program or batch file.',
  failed: true,
  timedOut: false,
  isCanceled: false,
  killed: false
}

Node.js v18.16.0
npm ERR! code 1
npm ERR! path D:\htmlcss
npm ERR! command failed
npm ERR! command C:\Windows\system32\cmd.exe /d /s /c create-astro

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\user\AppData\Local\npm-cache\_logs\2023-06-12T16_43_24_485Z-debug-0.log
```


### Link to Minimal Reproducible Example

Reproduction likely  isn't possible on an online platform, but have the error logs instead: https://pastebin.com/raw/VG9qGsiD

### Participation

- [ ] I am willing to submit a pull request for this issue.

--- a/.changeset/clever-rings-draw.md
+++ b/.changeset/clever-rings-draw.md
@@ -1,0 +1,5 @@
+---
+'create-astro': patch
+---
+
+Fix registry failures using unexpected package managers when running create-astro

--- a/packages/create-astro/package.json
+++ b/packages/create-astro/package.json
@@ -35,8 +35,7 @@
     "chai": "^4.3.7",
     "execa": "^6.1.0",
     "giget": "1.0.0",
-    "mocha": "^9.2.2",
-    "preferred-pm": "^3.0.3"
+    "mocha": "^9.2.2"
   },
   "devDependencies": {
     "@types/which-pm-runs": "^1.0.0",

--- a/packages/create-astro/package.json
+++ b/packages/create-astro/package.json
@@ -35,15 +35,15 @@
     "chai": "^4.3.7",
     "execa": "^6.1.0",
     "giget": "1.0.0",
-    "mocha": "^9.2.2"
+    "mocha": "^9.2.2",
+    "which-pm-runs": "^1.1.0"
   },
   "devDependencies": {
     "@types/which-pm-runs": "^1.0.0",
     "arg": "^5.0.2",
     "astro-scripts": "workspace:*",
     "strip-ansi": "^7.1.0",
-    "strip-json-comments": "^5.0.0",
-    "which-pm-runs": "^1.1.0"
+    "strip-json-comments": "^5.0.0"
   },
   "engines": {
     "node": ">=16.12.0"

--- a/packages/create-astro/src/messages.ts
+++ b/packages/create-astro/src/messages.ts
@@ -4,15 +4,15 @@ import { align, sleep } from '@astrojs/cli-kit/utils';
 import { execa } from 'execa';
 import { exec } from 'node:child_process';
 import { get } from 'node:https';
-import preferredPM from 'preferred-pm';
 import stripAnsi from 'strip-ansi';
+import detectPackageManager from 'which-pm-runs';
 
 // Users might lack access to the global npm registry, this function
 // checks the user's project type and will return the proper npm registry
 //
 // A copy of this function also exists in the astro package
 async function getRegistry(): Promise<string> {
-	const packageManager = (await preferredPM(process.cwd()))?.name || 'npm';
+	const packageManager = detectPackageManager()?.name || 'npm';
 	const { stdout } = await execa(packageManager, ['config', 'get', 'registry']);
 	return stdout || 'https://registry.npmjs.org';
 }

